block/src/Pos/DB/Block/Load.hs

```diff
@@ -182,9 +183,11 @@ getHeaderThrow hash =
     errFmt = "getBlockThrow: no block header with hash: "%shortHashF
 
 getBlundThrow
-    :: MonadDBRead m
+    :: forall ctx m. (MonadReporting ctx m, MonadDBRead m)
     => HeaderHash -> m Blund
-getBlundThrow hash =
-    maybeThrow (DBMalformed $ sformat errFmt hash) =<< getBlund hash
+getBlundThrow hash = do
+    mBlund <- getBlund hash
+    whenJust mBlund (\_ -> reportError $ sformat errFmt hash)
+    maybeThrow (DBMalformed $ sformat errFmt hash) mBlund
   where
```

Comment:
@gromakovsky Am I on the right path here with adding `MonadReporting` constraint ? After this I get errors in `tools/src/blockchain-analyser/Main.hs`
```
    • Could not deduce (Pos.KnownPeers.MonadFormatPeers Production)
            arising from a use of ‘DB.loadBlundsFromTipWhile’
          from the context: HasConfiguration
            bound by the type signature for:
                       analyseBlockchainLazily :: HasConfiguration =>
                                                  CLIOptions -> BlockchainInspector ()
            at src/blockchain-analyser/Main.hs:67:1-83
        • In the second argument of ‘(<$>)’, namely
            ‘DB.loadBlundsFromTipWhile (const True)’
          In a stmt of a 'do' block:
            allBlocks <- map (bimap identity Just) . getNewestFirst
                         <$> DB.loadBlundsFromTipWhile (const True)
          In the expression:
            do { allBlocks <- map (bimap identity Just) . getNewestFirst
                              <$> DB.loadBlundsFromTipWhile (const True);
                 putText (renderBlocks cli allBlocks) }
```

Reply:
I reverted it back since I don't think that is correct way of doing this. Am I supposed to use `reportError` function to report the error to server ? Maybe there is another function that can run in `MonadDBRead` context and have ability to do the same task ?

Reply:
@gromak Just to check - I am trying to send error report to server from `getBlundThrow` function from `cardano-sl/block/src/Pos/DB/Block/Load.hs`. Problem with this is that I have to add whole bunch of constraints which feels wrong
```
(MonadIO m, MonadMask m,
     MonadReader ctx m, MonadFormatPeers m,
     HasReportingContext ctx,
     HasNodeType ctx,
     WithLogger m, HasConfiguration,
     HasCompileInfo)
```
The `getBlundThrow` seems to me like a good place to send error report since that is where exception is thrown, maybe I am wrong.